### PR TITLE
Force encoding label to utf-8 for DOMString record data

### LINKED ISSUE #521: NDEFRecordInit#encoding should be only for BufferSource data source
https://w3c.github.io/web-nfc/#mapping-string-to-ndef needs to be refined.

If the data source is a string, we always get its `utf-8` bytes and write as the payload, i.e. `encoding` is forcibly set to "utf-8".

`NDEFRecordInit#encoding` is only used for WebDevs to declare how the `BufferSource` data is encoded.

### PR DESCRIPTION
**Pull request by beaufortfrancois**
FIX https://github.com/w3c/web-nfc/issues/521

***
<a href="https://pr-preview.s3.amazonaws.com/beaufortfrancois/web-nfc/pull/522.html" title="Last updated on Jan 8, 2020, 12:20 PM UTC (c36c67f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/522/ab85f1c...beaufortfrancois:c36c67f.html" title="Last updated on Jan 8, 2020, 12:20 PM UTC (c36c67f)">Diff</a>

### CODE REVIEW

**Comment by beaufortfrancois** [comment]
@leonhsl Does this PR address your concerns?

File: index.html [hunk 1]
```diff
@@ -2876,13 +2876,27 @@ <h3><dfn>Writing or pushing content</dfn></h3>
...
+              <ol>
+                <li>
+                  Let |encoding label:string| be "`utf-8`".
+                </li>
+              </ol>
+              <dt>{{BufferSource}}</dt>
+              <ol>
+                <li>
+                  Let |encoding label:string| be |record|'s <a>encoding</a> if
+                  it [= map/exists =], or "`utf-8`".
```
Comment (kenchris) [inline index.html]: orelse

File: index.html [hunk 2]
```diff
@@ -2876,13 +2876,27 @@ <h3><dfn>Writing or pushing content</dfn></h3>
...
+                </li>
+                <li>
+                  If |encoding label| is not equal to "`utf-8`", "`utf-16`",
+                  "`utf-16le`" or "`utf-16be`" [= exception/throw =] a
+                  {{TypeError}}.
```
Comment (kenchris) [inline index.html:2896]: TypeError seems wrong as it is not something you set...

Comment (kenchris) [reply index.html:2896]: Oh it is, probably makes sense to move this test before the other then

**Review by kenchris — APPROVED** [review]

File: index.html (hunk 1 above)
Comment (beaufortfrancois) [reply index.html]: Done

File: index.html (hunk 2 above)
Comment (beaufortfrancois) [reply index.html:2896]: Not sure what you mean. Which other test?

Comment (kenchris) [reply index.html:2896]: Not test, I meant line

Comment (beaufortfrancois) [reply index.html:2896]: How? This looks good to me.
 
image

Comment (kenchris) [reply index.html:2896]: yeah its fine.

**Comment by leonhsl** [comment]
LGTM and Thanks!

**Pull request merged**